GCI Task-Add small ship

Bidi92 opened the pull request:
<!-- Thanks for submitting a pull request for Destination Sol! :-)
Please fill in some brief details below about the PR.
If it contains source code please make sure to do everything in the pre pull request checklist first.
If you add unit tests we'll love you forever! -->

# Description
> Adds a small ship 'Bidi' to the game

# Testing
> Open a new game and select ship 'Bidi Tiny'
# Outstanding Work


# Notes

### Pre Pull Request Checklist:
<!-- When scanning the code with SonarLint, just don't introduce any new issues, and maybe even fix a few existing ones -->
- [ ] Code has been scanned with [SonarLint](http://www.sonarlint.org/intellij/)
- [ ] Methods have been annotated with @ Nullable and @ Nonnull ([Read more](https://github.com/MovingBlocks/DestinationSol/wiki/@Nullable-&-@Nonnull-Quickstart))
- [ ] There are no errors present in the project
- [ ] Code has been formatted and indented
- [ ] Methods have appropriate Javadoc ([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))

GooeyHub commented:
Can one of the admins please verify this patch?

vampcat commented:
Couple of small issues.

First, the ship icon has to be a 32x32 *white* sprite. Look at other icons for reference.
Second, you've essentially copied the gun from federal into your module. Not acceptable.
Third, the collision box for the ship is off. You'll have to generate the collision box via Box2D-Editor (or any other similar software) and replace the current one with that.

Other than that, it seems to work great.

Bidi92 commented:
Hi..Thanks for your input. I will work on the ship icon . The task requirements were to add a small ship only (no new weapon was mentioned) . Therefore I used the same items there on including the collision box. Please advise on this.

vampcat commented:
Yes, the task did not mention weapons. Which means I'm totally fine with you not adding any weapons - but it makes no sense to copy-paste a weapon from a different module. Just remove that gun from your module for now, and change all references of `bidi:dualBlaster` back to `federal:dualBlaster`.

Regarding the collision box : The collision box is a part of the ship. When you open your ship's json file, the `rigidBody: { }` part is the collision box, and tells the game what your ship's boundary is. Fixing this is a part of the task.

Bidi92 commented:
Hi.. Thanks for clarification and help. I will work on these accordingly

Bidi92 commented:
Hi
I have updated the module "Bidi" with the required changes:-

Ship icon updated 
Removed weapons from module and related references in code
Updated the collision box

Thanks

vampcat commented on modules/bidi/assets/items/textfile.txt:
```diff
@@ -0,0 +1 @@
+
```
Is there any reason to have this empty file?

Bidi92 replied:
No specific reason for text file. I was just learning how to use the editor.
Updated the icon file as per required format and pushed it using git command. 
Thanks for your help

P.S I was just learning how to use the web loader since I had used git command line before for other completed tasks :)

vampcat commented:
Since there is no real reason to have the empty textfile, I would request that you delete it.

That small issue not withstanding, I'm approving the task with hopes that you will fix this post-approval.

vampcat commented:
@Bidi92 I'm once again going to request you, stop using the web uploader. This makes the entire commit history a mess.

"Added a file"
"Deleted a file"
"Added a file"
"Deleted a file"

How am I supposed to understand anything from this? We, as mentors, put in time and effort to review your tasks. The least you can do is not make our job harder.

vampcat commented:
@GooeyHu: ok to test

GooeyHub commented:
Hooray Jenkins reported success with all tests good!
